jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/credentials/CodeWhispererClientAdaptor.kt

```diff
@@ -68,6 +70,13 @@ interface CodeWhispererClientAdaptor : Disposable {
 class CodeWhispererClientAdaptorImpl(override val project: Project) : CodeWhispererClientAdaptor {
     private val mySigv4Client by lazy { createUnmanagedSigv4Client() }
 
+    // Will purge once sigV4 client for CW is no longer needed
+    private val KProperty0<*>.isLazyInitialized: Boolean
+        get() {
+            isAccessible = true
+            return (getDelegate() as Lazy<*>).isInitialized()
+        }
+
```

Comment:
just curious is this change required  to call `this::mySigv4Client.isLazyInitialized`

Reply:
yes, and previously we have the risk of calling client.close() without it being initialized, hence this change.